### PR TITLE
lower case the email

### LINKED ISSUE #2057: 🐛 [Bug] Password-Reset/Registration Dialogue is Case-Sensitive
> [datenbrei](https://github.com/datenbrei) **Authored by [datenbrei](https://github.com/datenbrei)**
_<time datetime="2019-10-28T12:22:02Z" title="Monday, October 28th 2019, 1:22:02 pm +01:00">Oct 28, 2019</time>_
_Closed <time datetime="2019-11-07T08:52:48Z" title="Thursday, November 7th 2019, 9:52:48 am +01:00">Nov 7, 2019</time>_
---

## :bug: Bugreport
When users want to reset their passwords or want to register and they use emails with uppcercase in it, these can not be found and you get an email which says:  _You requested a password reset but unfortunately we couldn't find an account associated with your e-mail address. Did you maybe use another one when you signed up?_

wrong-email

### Steps to reproduce the behavior
1. Open reset Password dialogue
2. Type in your emial with upper case letters

Screenshot - 2019-10-29T171429 064

Screenshot - 2019-10-29T171929 698

### Expected behavior
The given email for reset should be converted to lowercase

### Version & Environment
any

### PR DESCRIPTION
**Pull request by Tirokk**
> [Kachulio1](https://github.com/Kachulio1) **Authored by [Kachulio1](https://github.com/Kachulio1)**
_<time datetime="2019-10-31T18:01:45Z" title="Thursday, October 31st 2019, 7:01:45 pm +01:00">Oct 31, 2019</time>_
_Closed <time datetime="2019-11-05T23:33:28Z" title="Wednesday, November 6th 2019, 12:33:28 am +01:00">Nov 6, 2019</time>_
---

## 🍰 Pull request

Sets the user email to lower case to avoid case sensitive mismatch 

### Issues

- None

### Todo

- [X] None

### CODE REVIEW

File: backend/src/schema/resolvers/passwordReset.js [hunk 1]
```diff
@@ -34,7 +34,8 @@ export default {
   Mutation: {
     requestPasswordReset: async (_parent, { email }, { driver }) => {
       const nonce = uuid().substring(0, 6)
-      return createPasswordReset({ driver, nonce, email })
+      const lowerCasedEmail = email.toLowerCase()
+      return createPasswordReset({ driver, nonce, lowerCasedEmail })
```
Comment (Tirokk) [inline backend/src/schema/resolvers/passwordReset.js:38]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-11-04T21:15:07Z" title="Monday, November 4th 2019, 10:15:07 pm +01:00">Nov 4, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -34,7 +34,8 @@ export default {
   Mutation: {
     requestPasswordReset: async (_parent, { email }, { driver }) => {
       const nonce = uuid().substring(0, 6)
-      return createPasswordReset({ driver, nonce, email })
+      const lowerCasedEmail = email.toLowerCase()
+      return createPasswordReset({ driver, nonce, lowerCasedEmail })
```


This is not going to work as `email` and not `lowerCaseEmail` gets desctructured here: https://github.com/Human-Connection/Human-Connection/blob/master/backend/src/schema/resolvers/passwordReset.js#L5

Comment (Tirokk) [reply backend/src/schema/resolvers/passwordReset.js:38]: > [Kachulio1](https://github.com/Kachulio1) **Authored by [Kachulio1](https://github.com/Kachulio1)**
_<time datetime="2019-11-04T21:19:19Z" title="Monday, November 4th 2019, 10:19:19 pm +01:00">Nov 4, 2019</time>_

---


Yep, good catch will fix that.

Comment (Tirokk) [reply backend/src/schema/resolvers/passwordReset.js:38]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-11-04T23:56:20Z" title="Tuesday, November 5th 2019, 12:56:20 am +01:00">Nov 5, 2019</time>_

---


https://github.com/Human-Connection/Human-Connection/pull/2118 :point_left:

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-11-05T23:33:28Z" title="Wednesday, November 6th 2019, 12:33:28 am +01:00">Nov 6, 2019</time>_

---


I'm going to close this in favour of #2118 hope you don't mind @Kachulio1 ?

**Review by Mogge — CHANGES_REQUESTED** [review]
We don't have automated tests for correct email delivery so we need to test this manually. I believe this is borken :thinking:

**Pull request closed without merging**